netbox/netbox/middleware.py

```diff
@@ -43,12 +44,30 @@ def __call__(self, request):
             login_url = f'{settings.LOGIN_URL}?next={parse.quote(request.get_full_path_info())}'
             return HttpResponseRedirect(login_url)
 
+        # Check if the language should be activated before the language cookie has been set.
+        # This will happen for instance on the first request after login.
+        # The language will have been already activated by the framework if the cookie exists.
+        if (
+            request.user.is_authenticated and
+            settings.LANGUAGE_COOKIE_NAME not in request.COOKIES and
+            hasattr(request.user, 'config')
+        ):
```

Comment:
Are we sure this is needed? IIRC the first authenticated response should always be a redirect, so I'd expect the language cookie should already be set.

Reply:
Your assumption is true when the login form or even some types of SSO is used, but not for `netbox.authentication.RemoteUserBackend`.

If you're using header-based authentication without that code fragment, you'll end up with the first page being displayed in the browser's default language, despite having a correct cookie.

Reply:
Ah, right.